### PR TITLE
Traitlets 4.1 API

### PR DESCRIPTION
**Pull request by SylvainCorlay**
This first PR is doing two things
- moving the `sync=True` and `**widget_serialization` into the `.tag()` call
- using the observe API for interact.
- update the example notebooks where custom widgets are made to use the new traitlets API and the unique "jupyter-js-widgets" entry point for the javascript.

There are many other changes coming for the new traitlets API but I don't want to do everything in the same PR.
One reason is that I noticed that some things which should be done at traitlets cross-validation are done at the observe stage, which *will* cause race conditions in the case where the cross-validation yields a trait error.

### CODE REVIEW

**Comment by minrk** [comment]
Don't forget to update requirements in setup.py

**Comment by SylvainCorlay** [comment]
Yeah, I did it in a previous PR.

**Comment by minrk** [comment]
Oops, didn't catch that. Thanks.

**Comment by SylvainCorlay** [comment]
This is ready to be looked at.

File: examples/notebooks/Custom Widget - Hello World.ipynb [hunk 1]
```diff
@@ -225,7 +226,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "You first need to import the `widget` and `manager` modules.  You will use the manager later to register your view by name (the same name you used in the back end).  To import the modules, use the `require` method of [require.js](http://requirejs.org/) (as seen below).\n"
+    "You first need to import the `jupyter-js-widgets` module. To modules, use the `require` method of [require.js](http://requirejs.org/) (as seen below).\n"
```
Comment (jdfreder) [inline examples/notebooks/Custom Widget - Hello World.ipynb]: Accidentally deleted the "To **import the **modules,"

Comment (SylvainCorlay) [inline examples/notebooks/Custom Widget - Hello World.ipynb]: ok, replacing with `to import modules.`

File: examples/notebooks/Date Picker Widget.ipynb [hunk 2]
```diff
@@ -817,21 +807,21 @@
    ]
   ],
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 2",
    "language": "python",
-   "name": "python3"
+   "name": "python2"
```
Comment (jdfreder) [inline examples/notebooks/Date Picker Widget.ipynb]: The example notebooks are all supposed to be Python 3

Comment (jdfreder) [inline examples/notebooks/Date Picker Widget.ipynb]: Luckily, I don't think you have to reopen and re-save every one.  Instead, you can probably get away with altering this by hand.

File: examples/notebooks/Custom Widget - Hello World.ipynb (hunk 1 above)
Comment (jdfreder) [inline examples/notebooks/Custom Widget - Hello World.ipynb]: Ah yes, that's better.

File: examples/notebooks/Date Picker Widget.ipynb (hunk 2 above)
Comment (SylvainCorlay) [inline examples/notebooks/Date Picker Widget.ipynb]: Fixed. There were a couple of notebooks that were already set to Python 2, which I reverted to python. All squashed in earlier commits.

**Comment by SylvainCorlay** [comment]
This is all ready.

**Comment by jdfreder** [comment]
Sweet!  Thanks

**Pull request merged**